sequencer/src/lib.rs

```diff
@@ -286,38 +290,46 @@ pub async fn init_node(
     let public_ip = IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1));
     let orchestrator_client = OrchestratorClient::new(validator_args, public_ip.to_string());
 
-    let (mut config, wait_for_orchestrator) = match persistence.load_config().await? {
+    let (config, wait_for_orchestrator) = match persistence.load_config().await? {
         Some(config) => {
             tracing::info!("loaded network config from storage, rejoining existing network");
             (config, false)
         }
         None => {
             tracing::info!("loading network config from orchestrator");
-            let config = orchestrator_client.get_config(public_ip.to_string()).await;
+            let mut config: NetworkConfig<VerKey, StaticElectionConfig> =
+                orchestrator_client.get_config(public_ip.to_string()).await;
+
+            config = orchestrator_client
+                .post_and_wait_all_public_keys(
+                    config.node_index,
+                    config.config.my_own_validator_config.public_key,
```

Comment:
Here, we're still using the key that the orchestrator originally generated for us, which is known to all nodes and not secure. What we've changed is, using this key is now voluntary.

We should add a `private_staking_key` option to the command line/environment variables defined in `sequencer/options.rs`, and use that key instead. That way, each node operator can generate and use their own private key, and they never have to reveal it to other nodes.

Reply:
@dailinsubjam